cooked-validators/src/Cooked/MockChain/BlockChain.hs

```diff
@@ -51,19 +52,21 @@ data MockChainError where
   FailWith :: String -> MockChainError
   OtherMockChainError :: (Show err, Eq err) => err -> MockChainError
 
+data MCEUnbalanceableError
+  = -- | The balancing wallet miss some value to pay what is needed to balance
+    -- the transaction.
+    MCEUnbalNotEnoughFunds Wallet PV2.Value
+  | -- | There is value to return to the balancing wallet but not enough to
+    -- fullfill the min ada requirement and there is not enough in additional
+    -- inputs to make it possible.
+    MCEUnbalNotEnoughReturning [PV2.TxOutRef] [PV2.TxOutRef] PV2.Value
```

Comment:
From a random user viewpoint, I thought that, regardless of the values, this kind of error would probably be addressed by the user by modifying the initial distribution of the balancing wallet to contain an arbitrary big amount of Ada more. I also thought that the user would refer to the TxOutRef printed in the rest of the trace if they really want more detail.
From an advanced user/dev viewpoint, I thought that displaying which utxos are involved could come in handy in case of something unexpected happening during the balancing itself.

I fully agree though that it would not hurt to have the Ada amounts contained in those sets of utxos in addition to or instead of the list of txoutrefs.